File: Leetcode 3001-3100/3069/3069.java
// Leetcode 3069: Distribute Elements Into Two Arrays I
// https://leetcode.com/problems/distribute-elements-into-two-arrays-i/
// Solved on 20th of August, 2026
class Solution {
    /**
     * Distributes elements of an array into two subarrays based on the greater 
     * value of their last added elements, then returns their concatenation.
     *
     * @param nums  The input integer array to be partitioned (minimum length of 2).
     * @return      An integer array containing all elements from the first 
     *              subarray followed by all elements from the second subarray.
     */
    public int[] resultArray(int[] nums) {
        int n = nums.length;
        int[] firstArray = new int[n];
        int[] secondArray = new int[n];

        firstArray[0] = nums[0];
        secondArray[0] = nums[1];

        int firstCount = 1;
        int secondCount = 1;

        for (int i = 2; i < n; i++) {
            if (firstArray[firstCount - 1] > secondArray[secondCount - 1]) {
                firstArray[firstCount++] = nums[i];
            } else {
                secondArray[secondCount++] = nums[i];
            }
        }

        int[] result = new int[n];
        int index = 0;
        for (int i = 0; i < firstCount; i++) {
            result[index++] = firstArray[i];
        }
        for (int i = 0; i < secondCount; i++) {
            result[index++] = secondArray[i];
        }

        return result;
    }
}

File: Leetcode 3001-3100/3069/3069.py
# Leetcode 3069: Distribute Elements Into Two Arrays I
# https://leetcode.com/problems/distribute-elements-into-two-arrays-i/
# Solved on 20th of August, 2026
class Solution:
    def resultArray(self, nums: list[int]) -> list[int]:
        """
        Distributes elements into two subarrays based on comparing their last appended
        elements and returns their concatenation.

        Parameters:
            nums (List[int]): An array of integers with a length of at least 2.

        Returns:
            List[int]: The concatenated array of the first subarray followed by the
                       second subarray.
        """
        fA = [nums[0]]
        sA = [nums[1]]

        for num in nums:
            if fA[-1] > sA[-1]:
                fA.append(num)
            else:
                sA.append(num)

        return fA + sA